tests/integration/test_csp.py

```diff
@@ -0,0 +1,59 @@
+import contextlib
+
+import pytest
+import flask_talisman
+from selenium.common.exceptions import NoSuchElementException
+
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+from dash.dependencies import Input, Output
+
+
+@contextlib.contextmanager
+def does_not_raise():
+    yield
+
+
+@pytest.mark.parametrize(
+    "add_hashes, hash_algorithm, expectation",
+    [
+        (False, None, pytest.raises(NoSuchElementException)),
+        (True, "sha256", does_not_raise()),
+        (True, "sha384", does_not_raise()),
+        (True, "sha512", does_not_raise()),
+        (True, "sha999", pytest.raises(ValueError)),
+    ],
+)
+def test_csp_hashes_inline_scripts(dash_duo, add_hashes, hash_algorithm, expectation):
```

Comment:
```suggestion
def test_incs001_csp_hashes_inline_scripts(dash_duo, add_hashes, hash_algorithm, expectation):
```

Just adding a test case ID for local targeting :)